### ISSUE TITLE
Terrascript - Dynamic block

### ISSUE THREAD

**Issue by Astrashh** [PolyhedralDev/Terra#75]
Allow the block argument for the Terrascript block function to be dynamic. Eg. picking a random ore type to be contained in a boulder would require duplicating all the block functions for each ore, which is unwieldy, but with dynamic blocks you could just set the block to a variable and change that variable in one place.

**Comment by dfsek** [comment]
This will be implemented as its own function, `dynamicBlock`, that way we can more easily cover the documentation in warnings.

**Comment by solonovamax** [comment]
alternative to that: what if we just implemented kotlin style switch cases?

```kotlin
when(randomInt(8)) {
    0 -> block(1, 2, 3, "minecraft:stone");
    1 -> block(1, 2, 3, "minecraft:coal_ore");
    2 -> block(1, 2, 3, "minecraft:iron_ore");
    3 -> block(1, 2, 3, "minecraft:gold_ore");
    4 -> block(1, 2, 3, "minecraft:redstone_ore");
    5 -> block(1, 2, 3, "minecraft:diamond_ore");
    6 -> block(1, 2, 3, "minecraft:emerald_ore");
    7 -> block(1, 2, 3, "minecraft:netherrite_block");
    else -> print("h o w");
}
```
it could also be really useful for the check function.
```kotlin
when(check(1, 2, 3)) {
    "AIR" -> print("is air");
    "OCEAN" -> print("is ocean");
    "LAND" -> {
        print("is land");
        block(1, 2, 3, "minecraft:stone");
    }
    else -> print("error");
}
```
they can also remove large if chains. For example, from the `stronghold.tesf` structure:
```kotlin
when {
    check( 13, y + 0, -8) != "LAND" -> fail;
    check(-12, y + 5, -8) != "LAND" -> fail;
    check(-12, y + 0,  8) != "LAND" -> fail;
    check( 13, y + 0,  8) != "LAND" -> fail;
    check( 13, y + 5, -8) != "LAND" -> fail;
    check( 13, y + 5,  8) != "LAND" -> fail;
    check(-11, y + 0, -8) != "LAND" -> fail;
    check(-12, y + 5,  8) != "LAND" -> fail;
}
```

**Comment by solonovamax** [comment]
Also, wouldn't you be able to do this?
```java
str blockType = "";
num block = randomInt(3);
if (block == 1)
    blockType = "minecraft:one";
else if (block == 2)
    blockType = "minecraft:two";
else if (block == 3)
    blockType = "minecraft:three";

block(1, 2, 3, blockType); // use as many of these as you want.
```

**Comment by dfsek** [comment]
Implemented as `dynamicBlock` in cda2d4688c6561fec23fdaa5538a55d9faee604d
